trie.py

```diff
@@ -0,0 +1,73 @@
+# https://www.codingninjas.com/studio/problems/implement-trie_631356?leftPanelTabValue=PROBLEM
+from sys import stdin, setrecursionlimit
+setrecursionlimit(10**7)
```

Comment:
Why system's recursion limit is being changed?

Reply:
this was given implictly in the question link mentioned above

Reply:
Pls write comment for it that why it's required

Reply:
Done

Reply:
This comment doesn't explain why such changes are present in code. Comment should be explaining the purpose of below code and where to use it, where to not use it
```
from sys import stdin, setrecursionlimit
setrecursionlimit(10**7)
```

Reply:
The recursion limit is set to 10**7  to prevent the program from running out of memory or encountering a stack overflow error.

Reply:
Running out of memory is mostly a good idea, you can't have your program taking more than a limited memory else it'll cause system failure. And same follows for stack overflow error unless you're very clear about pros and cons of it.

Reply:
By default, python limits max stack for recursion to 1000 recursive calls. You can confirm it by
```
import sys
print(sys.getrecursionlimit())
```
For this particular code, as all the operations are using **Iterative approach** it makes no difference but IF you were to use recursive calls for say insert or search (Example -> Search in reverse order) then with python's current limit of 1000 being stack limit.. Any word having char length more than 1000 would end up causing 1000+ recursive calls leading to program failure. If you implement any method using **Recursive approach** where valid computation can go beyond 1000 recursive calls (word having 1000+ chars for this example), you need to acc resize the system stack limit. This is all constraint based. This is one of many reasons why VALID DFS based questions dont contain testcases or constraints which can lead up to 1000 >= recursive calls